pkg/resources/manifests/deploy.yaml

```diff
@@ -239,12 +239,27 @@ spec:
                 # https://github.com/inspektor-gadget/inspektor-gadget/blob/3c51ff5e9f5b/gadgets/trace_dns/program.bpf.c#L365-L366
                 - NET_RAW
           volumeMounts:
-            - mountPath: /host
-              name: host
+            - mountPath: /host/bin
+              name: bin
+              readOnly: true
+            # We need to have read/write as we write NRI and OCI config files
+            # here.
+            - mountPath: /host/etc
+              name: etc
+            # We need to have read/write as we write NRI and OCI binaries here.
+            - mountPath: /host/opt
+              name: opt
+            - mountPath: /host/usr
+              name: usr
+              readOnly: true
+            - mountPath: /host/run
+              name: run
+              readOnly: true
+            - mountPath: /host/proc
+              name: proc
+              readOnly: true
```

Comment:
Does it actually force `/host/proc` to be readonly? I can still modify the **entire** host filesystem by using `/host/proc/1/cwd/`, like `rm /host/proc/1/cwd/bin/ping`. Can we prevent it on any way?

Reply:
@mauriciovasquezbernal I think the proc mount is read only, but that path contains a symlink and the destination is not read-only.

E.g:
```
$ ls -l /proc/self/cwd
lrwxrwxrwx 1 rodrigo rodrigo 0 Jan 30 15:11 /proc/self/cwd -> /home/rodrigo
```

My guess is that cwd (or root, also in that same dir) point to a directory that is not readOnly, and that is all.

Is it possible @eiffel-fl ? Can you print the output of `ls -l /proc/1/` and `mount` from that pod?

Be aware that `/bin/` might be a symlink too (like in a "merged" /usr, it can be a symlink to /usr/bin), so try to look all the paths for symlinks. My guess is that what it ends up pointing to is a a path mounted rw, and that is why you can write via /proc too.

Reply:
@eiffel-fl interesting. So you see this foobar file on the host too, if you just ssh to the host, right? Can you verify that, just in case? I would be surprised if you don't, but let's make sure when you claim you write to the host fs, you actually do it.

IMHO, if you write to the host fs, the most likely scenario is that there is a bizarre reason for the kubelet to do this, as weird mount namespace whatever, magic links in /proc being really magic and blah. But there is also the chance that this can be improved in upstream k8s.

The most likely scenario, as in the link that you shared, is that to mount the hostPath /, basically you are privileged and there will be no CVE associated due to a privilege container escaping. However, _maybe_ we can do better to prevent this.

I'd reduce this to a minimal example: leave alone the daemonset, just create a pod, remove as many mounts as possible, etc. And share the minimal repro here too, if you can :)

Reply:
> The thing is that we are no more `privileged: true` in Inspektor Gadget since aaaaaages: #348.
> But, we sadly still need to use `CAP_SYS_ADMIN`: #2228.

This is confusing, but privileged can refer not just to that field, but to the `privileged` PSS. If you need hostPath or cap sys_admin, you are privileged in the PSS sense. And that is what will probably won't grant you a CVE.

When baking the small repro, please try removing capabilities. I wouldn't be shocked if some "magic-links" in proc are not so magic if you lose some capabilities or things like that.

Reply:
I did a quick experiment, it's all about using `container.apparmor.security.beta.kubernetes.io/gadget: "unconfined"` with `SYS_PTRACE`, if I remove any of the two reading or writing /host/proc/1/root fails.

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: test-pod
  namespace: default
  annotations:
    container.apparmor.security.beta.kubernetes.io/gadget: "unconfined"
spec:
  hostPID: false
  hostNetwork: false
  terminationGracePeriodSeconds: 0
  restartPolicy: Never
  containers:
    - name: gadget
      terminationMessagePolicy: FallbackToLogsOnError
      image: ubuntu:22.04
      command: [ "sleep", "inf" ]
      securityContext:
        capabilities:
          drop:
            - ALL
          add:
            - SYS_PTRACE
      volumeMounts:
        - mountPath: /host/proc
          name: proc
          readOnly: true
  volumes:
    - name: proc
      hostPath:
        path: /proc
```

```bash 
$ kubectl exec -it test-pod -- bash -c "touch /host/proc/1/root/foo"
$ ls /foo
/foo
```

Reply:
@mauriciovasquezbernal interesting! I wonder what SYS_PTRACE gives you to allow for that, but very nice. Today I learned!